### PR TITLE
sprites for magiclysm goblins, orcs, and demon spiders.

### PR DESCRIPTION
**Pull request by phaethonfire**
#### Summary

No sprites existed for goblins, orcs, or demon spiders from the Magiclysm mod.  I have added sprites and tested locally.

#### Content of the change

9 new png sprites placed in both the Retrodays and NeoDays paths with associated json and credits.txt files.

#### Testing

I built the tilesheet using compose.py and spawned monsters in a local CDDA game.  Everything seemed to render okay.  First time contributor to a tileset.  Please let me know if you see anything nonstandard or problematic.

image

### CODE REVIEW

File: gfx/NeoDays/pngs_tiles_10x10/mod/magiclysm/monsters/goblins/goblins.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "id": "mon_goblin_warrior",
+    "fg": "mon_goblin_warrior"
```
Comment (Fris0uman) [inline gfx/NeoDays/pngs_tiles_10x10/mod/magiclysm/monsters/goblins/goblins.json:4]: Don't need the json if "id" and "fg" are the same, basically calling the png file with the id of the thing they represent is enough

File: gfx/NeoDays/pngs_tiles_10x10/mod/magiclysm/monsters/goblins/credits.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+phaethonfire
```
Comment (Fris0uman) [inline gfx/NeoDays/pngs_tiles_10x10/mod/magiclysm/monsters/goblins/credits.txt:1]: don't need credit.txt if you're adding the sprite yourself, the credits will be in the commit history

**Review by Fris0uman — COMMENTED** [review]
You can remove the json and txt that are not usefull, and also add a screenshot of those sprite in game in the description of the pull request, in #Testing part

File: gfx/NeoDays/pngs_tiles_10x10/mod/magiclysm/monsters/goblins/goblins.json (hunk 1 above)
Comment (phaethonfire) [reply gfx/NeoDays/pngs_tiles_10x10/mod/magiclysm/monsters/goblins/goblins.json]: got it.  I matched all png filenames with the monsters they represented, so I will remove the json documents.

File: gfx/NeoDays/pngs_tiles_10x10/mod/magiclysm/monsters/goblins/credits.txt (hunk 2 above)
Comment (phaethonfire) [reply gfx/NeoDays/pngs_tiles_10x10/mod/magiclysm/monsters/goblins/credits.txt]: sounds good.  I removed those files now.

**Comment by Fris0uman** [comment]
Almost ready, Retrodays sprite don't have transparent background they have black (0.0.0) backgrounds

**Comment by phaethonfire** [comment]
Thanks for all the careful attention, @Fris0uman.  Sorry I missed the background on Retrodays.

I fixed the transparency, tested building tiles with compose.py, and then loaded a game with Retrodays.  I think I may have actually done what I was supposed to this time.  Here is the conversion script that ChatGPT and I made to make the transparent pixels black:

```
import argparse
import cv2
import numpy as np
import os


def process_image(input_path, output_path):
    # Load the input image
    im = cv2.imread(input_path, -1)

    # Replace transparent pixels with black
    im[np.where(im[:, :, 3] == 0)] = (0, 0, 0, 255)

    # Write the output image
    cv2.imwrite(output_path, im)


if __name__ == '__main__':
    # Parse the command-line arguments
    parser = argparse.ArgumentParser(description='Replace transparent pixels with black')
    parser.add_argument('input', help='Path to the input PNG file')
    parser.add_argument('output', help='Path to save the output PNG file')
    args = parser.parse_args()

    # Call the process_image function with the input and output file paths
    input_path = os.path.abspath(args.input)
    output_path = os.path.abspath(args.output)
    process_image(input_path, output_path)
```

And then Docker to satisfy dependencies and do the conversion.

`docker run --rm -v $PWD:/tilesets opencvcourses/opencv-docker:latest python /tilesets/transparent_to_black.py /tilesets/${file} /tilesets/${file}`

**Pull request merged**